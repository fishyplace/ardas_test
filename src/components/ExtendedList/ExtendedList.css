.ext-list {
    overflow: scroll;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translate(-50%, -50%);
    width: 1140px;
    height: 900px;
}
th, td {
    width: 100px;
    height: 50px;
    display: inline-block;
    margin: 0 50px;
    text-align: right;
}
.important td:first-child a {
    text-decoration: none;
    text-transform: uppercase;
    color: red;
}