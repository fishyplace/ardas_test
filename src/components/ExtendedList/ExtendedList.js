import React from 'react';
import './ExtendedList.css';
import { Link } from 'react-router-dom';

const extendedList = props => {
    const {data} = props;
    let toDisplay = null;

    if (data.length !== 0) {
        toDisplay = data.map(el => (
            <tr key={el.id} className={el.is_high_priority ? 'important' : null}>
                <td>
                    <Link to={`/edit/${el.id}`}>
                        {el.name}
                    </Link>
                </td>
                <td>{el.estimated_effort}</td>
                <td>{el.actual_effort}</td>
                <td>{el.due_date ? new Date(el.due_date).toString().split(/[0-9]+:[0-9]+:[0-9]+/)[0] : null}</td>
                <td>{el.tags ? el.tags.join(' , ') : 'none'}</td>
            </tr>
            )
        )
    }

    return (
        <div className="ext-list">
            <table>
                <thead>
                    <tr>
                        <th>Name</th>
                        <th>Estimated</th>
                        <th>Actual</th>
                        <th>Due Date</th>
                        <th>Tags</th>
                    </tr>
                </thead>
                <tbody>
                    {toDisplay}
                </tbody>
            </table>
        </div>
    )
}

export default extendedList;