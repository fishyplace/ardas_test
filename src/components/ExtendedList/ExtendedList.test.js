import React from 'react';
import ExtendedList from './ExtendedList';

import { shallow } from 'enzyme';

it('should render with one prop object without crashing', () => {
    const data = [{
        id: 1,
        name: "Today_task1",
        creation_date: "2015-04-21T06:50:21",
        due_date: "2015-04-22T23:59:00",
        start_date: "2015-04-21T00:00:01",
        is_completed: false,
        is_archived: false,
        estimated_effort: 5.5,
        actual_effort: 3.3,
        physical_progress: 60,
        obj_status: "active",
        description: "Lorem_ipsum_dolor_sit_amet,_consectetur_adipiscing_elit",
        project_id: 0
      }];

      const component = shallow(<ExtendedList data={data} />);
      expect(component).toMatchSnapshot();
})