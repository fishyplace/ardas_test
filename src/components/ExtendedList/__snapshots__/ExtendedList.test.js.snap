// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with one prop object without crashing 1`] = `
<div
  className="ext-list"
>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Estimated
        </th>
        <th>
          Actual
        </th>
        <th>
          Due Date
        </th>
        <th>
          Tags
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className={null}
        key="1"
      >
        <td>
          <Link
            replace={false}
            to="/edit/1"
          >
            Today_task1
          </Link>
        </td>
        <td>
          5.5
        </td>
        <td>
          3.3
        </td>
        <td>
          Wed Apr 22 2015 
        </td>
        <td>
          none
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
